src/back/wgsl/writer.rs

```diff
@@ -185,7 +207,14 @@ impl<W: Write> Writer<W> {
         func_ctx: &FunctionCtx<'_>,
     ) -> BackendResult {
         let func_name = match func_ctx.ty {
-            FunctionType::EntryPoint(index) => self.names[&NameKey::EntryPoint(index)].clone(),
```

Comment:
I don't understand, what was wrong with the original code?

Reply:
Sorry, I should had to describe why we need these changes first. It's all about readability (except math function).

Before:
```wgsl
fn main1([[location(1)]] a_uv1: vec2<f32>, [[location(0)]] a_pos1: vec2<f32>) -> type10 {
```

After:
```wgsl
fn vs_main([[location(1)]] a_uv1: vec2<f32>, [[location(0)]] a_pos1: vec2<f32>) -> VertexOutput {
```

Now it's easier to merge fragment and vertex shaders generated from spv into one wgsl file (otherwise all entry points functions named `main`).

Reply:
Ideally I think we'd want to try to use the name of the entry point/types from SPIR-V instead of renaming it based on the stage.

There's also a risk if there are multiple entry points that use the same stage though I don't know if we currently allow that. e .g. multiple compute entry points in one file would collide

Reply:
I use multiple compute entry points in single WGSL files quite a lot. I’d also like to retain the ability to go wgsl-in -> process -> wgsl-out.

Reply:
@grovesNL @acowley Thank you for your comments! I removed this part from PR.